### DISCUSSION TITLE
Start & Stop listeners in an easy way

### DISCUSSION THREAD

**Discussion by kurtulekrem** [rabbitmq/rabbitmq-server#3788] [Ideas]
Hello,

There was a necessity to close default AMQP listener on 5672 permanently because I am using SSL AMQP listener on port 5671 so there is no need 5672 to stay up & running. Then I have set "listeners.tcp = none" in the configuration file and restart the RabbitMQ instances. It is worked and the problem is solved but I think this process should be done in a much easier way.

In Websphere MQ, it can be done by entering a command in order to start or stop a listener, or it can be done from Web User Interface. I think this process should also be valid for RabbitMQ. Users should not need to restart their RabbitMQ instances just to start or stop a listener.

I hope there will be some improvements about this process in the next releases.

Thanks and regards,

Stay safe.

**Comment by lukebakken** [comment]
In general users determine their RabbitMQ configuration via reading the docs and trial-and-error in a pre-production environment, then save that configuration (hopefully in source control) to be used in a production environment. Enabling / disabling listeners is something that should be done in configuration prior to starting your production RabbitMQ instance and should rarely, if ever, change.
